flik/linesearch/conditions.py

```diff
@@ -1 +1,47 @@
 """Conditions for the line search algorithm."""
+
+import numpy as np
+
+def check_armijo(func, grad, veck, desck, alpha, cone=1e-4):
+    r"""
+    Check if Armijo condition is satisfied for certain alpha.
+    A sufficient decrease is measured by:
+    :math:`f(x_k + \alpha s_k) - f(x_k) \leq c1 \alpha \nabla f(x_k) s_k`
+
+    Parameters:
+    -----------
+    func: function
+        Objective function
+    grad: function
+        Function to evaluate Gradient of the function at k-iteration
+    veck: np.ndarray((N,)), for N variables
+        Value of variables at k-iteration
+    desck: np.ndarray((N,)) for N variables
+        Descent direction
+    alpha: float
+        Step length
+    cone: float
+        Condition constant
+    Returns: bool
+    True if the condition is satisfied, False otherwise
+    """
+    # Parameter's checks
+    if not isinstance(veck, (np.ndarray, float)):
+        raise TypeError("Variable vector should be provided as a numpy ndarray of float")
+    if not isinstance(desck, (np.ndarray, float)):
+        raise TypeError("The descent direction vector should be provided as a numpy\
+                         ndarray of float")
+    if not isinstance(alpha, float):
+        raise TypeError("Alpha should be provided as a float")
+    if veck.size != desck.size:
+        raise ValueError("The size of xk and sk should be the same")
+    if alpha < 0. or alpha > 1.0:
+        raise ValueError("Alpha should have a value between 0. and 1.")
+
+    # Evaluate the function and gradient
+    funck = func(veck)
+    gradk = grad(veck)
+    fstep = func(veck + alpha*desck)
+
+    # Check the condition
+    return bool((fstep - funck <= cone*alpha*np.dot(desck, gradk)).all())
```

Comment:
I'm not too sure which way we will lean on this, but I think `desck.dot(gradk)` is nicer than the  `np.dot(desck, gradk)`. It doesn't matter now, but if we are multiplying more than two arrays, `a.dot(b).dot(c)` looks a lot cleaner than `np.dot(a, np.dot(b, c))`. It does, however, assume that the first object (`desck`) is a `np.ndarray` instance (or more specifically, contains the method `dot`), which ins't a problem in this case because we explicitly test that `desck` is a numpy array. `np.dot` is convenient because it can handle a wide range of inputs (see https://docs.scipy.org/doc/numpy/reference/generated/numpy.dot.html), so I'm not sure.

Reply:
so... should we use np.dot or not? I don't mind either...